### DISCUSSION TITLE
Open-sourcing cartesius

### DISCUSSION THREAD

**Discussion by ghost** [TeamSPWK/cartesius#26] [General]
@iycho-swk @iglee-swk @clement-swk 

Currently `cartesius` contains both the benchmark and the models implementation.

If we open-source it, people might re-use idea of others and it might get more difficult to evaluate applicants...

---

Should we separate the repository into 2 : one for the benchmark (maybe with a baseline model) and one for the models ?

**Comment by iycho-swk** [comment]
Perfect idea.

**Reply by iycho-swk** [reply]
One for the public (with a simple baseline model) which is linked to the kaggle.
The other is for private solely for us.

**Comment by clement-swk** [comment]
How do we easily maintain 2 versions of cartesius?

**Reply by ghost** [reply]
@clement-swk We will have 2 repo :

* `cartesius` [public] : contains code for polygon generation, data for validation / test sets, links to Kaggle competition, baseline notebook.
* `cartesius-admin` [private] : contains code to generate solution files of Kaggle competition, various notebooks for each models to reproduce scores.

Let me know your opinion about this !

**Reply by clement-swk** [reply]
Will we need to manually duplicate every modification we do to cartesius-admin to cartesius manually?

**Reply by ghost** [reply]
No, the 2 repo will be 2 different things.

Basically `cartesius` will be a package usable by anyone.

And `cartesius-admin` will just be a collection of notebooks, and scripts to help manage the Kaggle competitions.

**Reply by clement-swk** [reply]
Okay so cartesius [public] will be "bare bone" cartesius-admin right?

**Reply by ghost** [reply]
`cartesius` will be like the current repo, but without model implementation. Just the data part. We will add a baseline notebook so people can have a base score.

And we will move the model implementation into notebooks, which will be versioned in `cartesius-admin`.

---

Side note :

We will not have anymore the possibility to import models and reuse it easily in downstream task. It will become only a benchmark (for now).

I think it will make it easier to just throw some new implementation in a notebook and benchmark it. Later if we need to have centralized model implementation, we will see.
